modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/FieldMappingInfoResourceImpl.java

```diff
@@ -141,6 +151,27 @@ private void _addFieldMappingInfos(
 		}
 	}
 
+	private String _getIndexName(boolean external, String indexName) {
+		String fullIndexName = _indexNameBuilder.getIndexName(
+			contextCompany.getCompanyId());
+
+		if (external) {
+			fullIndexName = "external-" + indexName;
+		}
+		else if (!Validator.isBlank(indexName)) {
+			fullIndexName = fullIndexName + StringPool.DASH + indexName;
+		}
```

Comment:
I think it would be better to have blank check before if-else, because now the first if can still have null. Maybe: 
`		if (Validator.isBlank(indexName)) {
			return fullIndexName;
		}

		if (external) {
			fullIndexName = "external-" + indexName;
		}
		else {
			fullIndexName = fullIndexName + StringPool.DASH + indexName;
		}`

Reply:
Nice catch!